tests/basic/storage_pool_test.go

```diff
@@ -5794,3 +5794,196 @@ func addNewPools(n node.Node, numPools int) error {
+		log.InfoD("IO Priority of Pool [%s] before Pool expand is [%s]", poolUUID, ioPriorityBefore)
+
+		// Change IO Priority of the Pool
+		nodeDetail, err := GetNodeWithGivenPoolID(poolUUID)
+		log.FailOnError(err, "Failed to get Node Details using PoolUUID [%v]", poolUUID)
+
+		log.InfoD("Bring Node to Maintenance Mode")
+		log.FailOnError(Inst().V.EnterMaintenance(*nodeDetail), fmt.Sprintf("Failed to bring Pool [%s] to Mainteinance Mode on Node [%s]", poolUUID, nodeDetail.Name))
+
+		// Set IO Priority on the Pool
+		var ioPriorities = []string{"low", "medium", "high"}
+		var setIOPriority string
+
+		// Selecting Pool IO Priority Value different that the one already set
+		for _, eachIOPriority := range ioPriorities {
+			if eachIOPriority != ioPriorityBefore {
+				setIOPriority = eachIOPriority
+				break
+			}
+		}
+
+		log.InfoD("Setting Pool [%s] with IO Priority [%s]", poolUUID, setIOPriority)
+		log.FailOnError(Inst().V.UpdatePoolIOPriority(*nodeDetail, poolUUID, setIOPriority), fmt.Sprintf("Failed to set IO Priority of Pool [%s]", poolUUID))
+
+		log.InfoD("Bring Node out of Maintenance Mode")
+		t := func() (interface{}, bool, error) {
+			if err := Inst().V.ExitMaintenance(*nodeDetail); err != nil {
+				return nil, true, err
+			}
+			return nil, false, nil
+		}
+
+		_, err = task.DoRetryWithTimeout(t, 15*time.Minute, 2*time.Minute)
+		log.FailOnError(err, fmt.Sprintf("Failed to exit maintenence mode on node [%s]", nodeDetail.Name))
+
+		// Do Pool Expand on the Node
+		stepLog = fmt.Sprintf("Expanding pool on node [%s] and pool UUID: [%s] using auto", nodeDetail.Name, poolUUID)
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			poolToBeResized, err := GetStoragePoolByUUID(poolUUID)
+			log.FailOnError(err, "Failed to get pool using UUID [%s]", poolUUID)
+			expectedSize := poolToBeResized.TotalSize * 2 / units.GiB
+
+			isjournal, err := isJournalEnabled()
+			log.FailOnError(err, "Failed to check if Journal enabled")
+
+			log.InfoD("Current Size of the pool [%s] is [%d]", poolUUID, poolToBeResized.TotalSize/units.GiB)
+			err = Inst().V.ExpandPool(poolUUID, api.SdkStoragePool_RESIZE_TYPE_AUTO, expectedSize)
+			dash.VerifyFatal(err, nil, "Pool expansion init successful?")
+
+			resizeErr := waitForPoolToBeResized(expectedSize, poolUUID, isjournal)
+			dash.VerifyFatal(resizeErr, nil, fmt.Sprintf("Verify pool [%s] on node [%s] expansion using auto", poolUUID, nodeDetail.Name))
+		})
+
+		// Validate if PoolIO Priority is not changed after pool Expansion
+		ioPriorityAfter, err := GetPoolIOPriorityStatus(poolUUID)
+		log.FailOnError(err, "Failed to get IO Priority for Pool with UUID [%v]", poolUUID)
+
+		log.InfoD(fmt.Sprintf("Priority Before [%s] and Priority after Pool Expansion [%s]", ioPriorityBefore, ioPriorityAfter))
+
+		opts := make(map[string]bool)
+		opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+		ValidateAndDestroy(contexts, opts)
```

Comment:
can you add the apps destroy as defer after validate applications

Reply:
done